File: ClothingStoreApp/ClothingStore.Application/DTOs/AvailableLocationBySizeDto.cs
namespace ClothingStore.Application.DTOs
{
    public record class AvailableLocationBySizeDto
    {
        public Guid AvailableLocationBySizeId { get; set; }
        public Guid VariantId { get; set; }
        public Guid SizeId { get; set; }
        public IEnumerable<StockByLocationDto> AvailableLocationsOfGivenSize { get; set; } = Array.Empty<StockByLocationDto>();
    }
}

File: ClothingStoreFrontend/src/Types/AvailableLocationBySizeDto.ts
import type { AnySizeDto } from "./SizeDto";
import type { StockByLocationDto } from "./StockByLocationDto";

export interface AvailableLocationBySizeDto {
    availableLocationBySizeId: string;
    variantId: string;
    size: AnySizeDto;

    availableLocationsOfGivenSize: StockByLocationDto[];
}